lib/elixir/lib/module/parallel_checker.ex

```diff
@@ -158,8 +159,50 @@ defmodule Module.ParallelChecker do
     :gen_server.call(server, {:unlock, module})
   end
 
+  defp prepare_modules(modules) do
+    Enum.map(modules, fn {map, binary} ->
```

Comment:
We should either: only load it for the runtime modules or have the parallel compiler load the .beam for runtime modules.

Reply:
We need to load it for compiled modules when we add the ExCk chunk and rebuild it. Now we load it for all modules at this location and pass it along til we add the chunk.

Reply:
My concern is that we are doing this for all modules. We should load it only for the runtime ones. Maybe we need a more efficient mechanism to recheck runtime warnings.

Reply:
We are doing it for runtime dependent modules and compiled modules. For runtime dependent modules we need to do get information we usually get from the module map, for compiled modules we need to do it to add the ExCk chunk.

Right now we update the ExCk chunk in runtime dependent modules, which is not necessary yet, but will be necessary when the type inference gets more advanced.